tests/test_v_configuration.py

```diff
@@ -0,0 +1,29 @@
+from unittest.mock import patch, MagicMock
+from streamlit.testing.v1 import AppTest
+
+def sql_handler(*args):
+    query_result = MagicMock()
+    stmt = ' '.join(args[0].split())
+
+    if stmt == 'select is_first_time_setup_dismissed from config_data.configuration':
+        query_result.collect.return_value = [{'IS_FIRST_TIME_SETUP_DISMISSED': True}]
+    elif stmt == 'select enable_warning_generation_task from config_data.configuration limit 1':
+        query_result.collect.return_value = [{'ENABLE_WARNING_GENERATION_TASK': True}]
+    else:
+        raise NotImplementedError(f'"{stmt}"')
+    
+    return query_result
+
+@patch('snowflake.snowpark.session.Session.builder')
+def test_streamlit_configuration_ui(builder):
+    session = MagicMock()
+    builder.getOrCreate.return_value = session
+    session.sql.side_effect = sql_handler
```

Comment:
Thoughts on that pattern? I generally prefer having the test be self-contained and specifying expectations directly in the test, especially with lots of test cases in a file.

Reply:
I can move def sql_handler to be within the test. Is that what you meant?

```
@patch('snowflake.snowpark.session.Session.builder')
def test_streamlit_configuration_ui(builder):
    session = MagicMock()
    builder.getOrCreate.return_value = session
    def sql_handler(*args):
         ...
    session.sql.side_effect = sql_handler
```

Reply:
Wondering what the rest of the team thinks here, this one is highly subjective.

Reply:
If you don't feel strongly about this, I will keep it as is - the reasoning is that we have one test per file, so it's nice to have a bit of organization, as the sql part could get lengthy.

Reply:
Yeah, the lack of the ability to register individual mocks means this is what e.g. UDFs and sprocs would look like too if Snowpark wasn't building the registry. I think it's _slightly_ better inside of the same function, but given there's only one test and it's short, either way is clear to me.

As a general pattern I think your comment @sfc-gh-melnacouzi is better.